pkg/informant/endpoints.go

```diff
@@ -46,13 +113,64 @@ func (s *State) RegisterAgent(info *api.AgentDesc) (*api.InformantDesc, int, err
 //
 // Returns: body (if successful), status code and error (if unsuccessful)
 func (s *State) TryDownscale(target *api.RawResources) (*api.DownscaleResult, int, error) {
-	return &api.DownscaleResult{Ok: true, Status: "No action taken"}, 200, nil
+	// If we aren't interacting with a cgroup, then we don't need to do anything.
+	if s.cgroup == nil {
+		return &api.DownscaleResult{Ok: true, Status: "No action taken (no cgroup enabled)"}, 200, nil
+	}
+
+	requestedMem := uint64(target.Memory.Value())
+	newMemHigh := s.cgroup.config.calculateMemoryHighValue(requestedMem)
+	ok, status, err := func() (bool, string, error) {
+		s.cgroup.updateLock.Lock()
+		defer s.cgroup.updateLock.Unlock()
+
+		mib := float64(1 << 20) // 1 MiB = 2^20 bytes
+
+		current, err := s.cgroup.getCurrentMemory()
+		if err != nil {
+			return false, "", fmt.Errorf("Error fetching getting cgroup memory: %w", err)
+		}
+
+		// For an explanation, refer to the documentation of CgroupConfig.MemoryHighBufferBytes
+		//
+		// TODO: this should be a method on (*CgroupConfig).
+		if newMemHigh < current+s.cgroup.config.MemoryHighBufferBytes {
+
+			verdict := "Calculated memory.high too low"
+			status := fmt.Sprintf(
+				"%s: %v MiB (new high) < %v MiB (current usage) + %v MiB (buffer)",
+				verdict,
+				float64(newMemHigh)/mib, float64(current)/mib,
+				float64(s.cgroup.config.MemoryHighBufferBytes)/mib,
+			)
+
+			return false, status, nil
+		}
+
+		// Downscale:
+		//
+		// TODO: see similar note above. We shouldn't call methods on s.cgroup.mgr from here.
+		if err := s.cgroup.mgr.SetHighMem(newMemHigh); err != nil {
+			return false, "", fmt.Errorf("Error setting cgroup memory.high: %w", err)
+		}
+
+		status := fmt.Sprintf("Set cgroup memory.high down to %v MiB", float64(newMemHigh)/mib)
```

Comment:
%f for floats?

Reply:
%v because it doesn't include decimals if it doesn't need to, which is _almost_ guaranteed to be the case.

For floats, %v is equivalent to %g - might make sense to use that instead tbh.

Reply:
Changed to %g.